### PR TITLE
codeintel: RFC 235: Add postgres-backed store implementation

### PR DESCRIPTION
**Pull request by efritz**
A step towards [RFC 235: Move LSIF data to Postgres](https://docs.google.com/document/u/1/d/1Y9p29hK8xrPUTvBdWqP9uAHDg3JLV-HCaxsbCo9-YHQ). This a prerequisite to https://github.com/sourcegraph/sourcegraph/issues/13886, https://github.com/sourcegraph/sourcegraph/issues/13888, and https://github.com/sourcegraph/sourcegraph/issues/13889.

This PR adds a bundle store implementation that targets Postgres, but does not update any place where SQLite is currently used. These need to be added in a certain order to ensure we do not write data to a place we're not querying. **Do not** use this new code until https://github.com/sourcegraph/sourcegraph/issues/13887 has been completed and deployed.

### CODE REVIEW

File: enterprise/internal/codeintel/bundles/persistence/postgres/batch.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	if batch := i.pop(); len(batch) != 0 {
+		// ($xxxxx,$xxxxx,...)
+		//  ^^^^^^^ * n - 1 (extra comma) + 2 (parens)
+		rowSize := (7*i.numColumns + 1)
+
+		// Determine number of rows being flushed
+		numRows := len(batch) / i.numColumns
+
+		// Account for commas separating rows
+		n := numRows*rowSize + (numRows - 1)
```
Comment (Strum355) [inline enterprise/internal/codeintel/bundles/persistence/postgres/batch.go:69]: As someone who's no idea the shape of the data :exploding_head:

Comment (efritz) [reply enterprise/internal/codeintel/bundles/persistence/postgres/batch.go]: Refactored to make this more clear :)

**Review by Strum355 — APPROVED** [review]
:bread:

**Comment by chrismwendt** [comment]
Looks like batch insertion was added in this PR. I'm curious what kind of perf improvement you saw from batching, if you have some benchmark numbers or anecdotes about it.

**Pull request merged**